tests/creditNote.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Test\CreditNote;
+use \Exception;
+use StarkInfra\CreditNote;
+use StarkInfra\Transfer;
+use StarkInfra\Invoice;
+use StarkInfra\Signer;
+
+class TestCreditNote
+{
+    public function createAndCancel()
+    {
+        $creditNote = CreditNote::create([TestCreditNote::exampleCCB(null)])[0];
+
+        if (is_null($creditNote->id)) {
+            throw new Exception("failed");
+        }
+
+        $creditNoteCan = CreditNote::cancel($creditNote->id);
+
+        if (is_null($creditNoteCan->id)) {
+            throw new Exception("failed");
+        }
+    }
+
+    public function queryAndGet()
+    {
+        $creditNotes = iterator_to_array(CreditNote::query(["limit" => 10]));
+        
+        if (count($creditNotes) != 10) {
+            throw new Exception("failed");
+        }
+
+        $creditNote = CreditNote::get($creditNotes[0]->id);
+
+        if ($creditNotes[0]->id != $creditNote->id) {
+            throw new Exception("failed");
+        }
+    }
+
+    public function getPage()
+    {
+        $ids = [];
+        $cursor = null;
+        for ($i=0; $i < 2; $i++) { 
+            list($page, $cursor) = CreditNote::page($options = ["limit" => 5, "cursor" => $cursor]);
+            foreach ($page as $creditNote) {
+                if (in_array($creditNote->id, $ids)) {
+                    throw new Exception("failed");
+                }
+                array_push($ids, $creditNote->id);
+            }
+            if ($cursor == null) {
+                break;
+            }
+        }
+        if (count($ids) != 10) {
+            throw new Exception("failed");
+        }
+    }
+
+    public function exampleCCB($schedule=false)
+    {
+
+        $params = [
+            "templateId" => "5707012469948416",
+            "name" => "Jamie Lannister",
+            "taxId" => "012.345.678-90",
+            "nominalAmount" => 100000,
+            "scheduled" => "2022-05-11",
+            "payment" => new Transfer(array(
+                    "bankCode" => "60701190",
+                    "branchCode" => "7248",
+                    "accountNumber" => "5005482-1",
+                    "taxId" => "594.739.480-42", 
+                    "name" => "Jamie Lannister"
+            )),
+            "paymentType" => "transfer",
+            "invoices" =>[
+                new Invoice(array(
+                    "amount" => 120000,
+                    "due" => "2022-07-11",
+                    "fine" => 3.0,
+                    "interest" => 1.0
+            ))
+        ], 
+            "signers" => [
+                new Signer(array(
+                    "contact" =>  "jamie.lannister@gmail.com",
+                    "method" => "link",
+                    "name" => "Jamie Lannister",
+            ))
+        ],
+            "rebateAmount" => 0,
+            "tags" => [
+                'War supply',
+                'Invoice #1234'
+            ],
+            "externalId" => "my_unique_id2012",
```

Comment:
Usa um gerador de id aqui pra não ter que trocar na mão toda vez que for rodar o teste, no pix request tem um exemplo:
"php-" . $uuid = mt_rand(0, 0xffffffff),